pandas/tseries/index.py

```diff
@@ -827,16 +828,21 @@ def _format_native_types(self, na_rep=u('NaT'),
     def to_datetime(self, dayfirst=False):
         return self.copy()
 
-    def astype(self, dtype):
+    def astype(self, dtype, copy=True):
         dtype = np.dtype(dtype)
```

Comment:
yes needs to be added to all implementations.
for most indexes you can simply pass this into the numpy astype directly

```
[Sat May 21 20:05:12 ~/pandas]$ grin 'def astype' pandas/indexes/
pandas/indexes/base.py:
  757 :     def astype(self, dtype):
pandas/indexes/multi.py:
 2077 :     def astype(self, dtype):
pandas/indexes/numeric.py:
  241 :     def astype(self, dtype):
[Sun May 22 08:49:08 ~/pandas]$ grin 'def astype' pandas/tseries/
pandas/tseries/index.py:
  830 :     def astype(self, dtype):
pandas/tseries/period.py:
  389 :     def astype(self, dtype):
pandas/tseries/tdi.py:
  438 :     def astype(self, dtype):
```

e.g. here's what the impl in ``numeric.py`` becomes
```
    def astype(self, dtype, copy=False):
        dtype = pandas_dtype(dtype)
        if is_float_dtype(dtype) or is_integer_dtype(dtype):
            values = self._values.astype(dtype, copy=copy)
        elif is_object_dtype(dtype):
            values = self._values.astype('object', copy=copy)
        else:
            raise TypeError('Setting %s dtype to anything other than '
                            'float64 or object is not supported' %
                            self.__class__)
        return Index(values, name=self.name, dtype=dtype)
```